python/pyspark/sql/types.py

```diff
@@ -1160,18 +1150,22 @@ def _has_struct_or_date(dt):
 def _create_properties(fields):
     """Create properties according to fields"""
```

Comment:
we need to expand this comment as well to more clearly define what is happening with this function.